SCR/scr/tests/test_gp_summary_upload.py

```diff
@@ -108,5 +108,41 @@ def test_json_string_example(self):
         json_file = str(self.hashFileDir / 'hash16UK05.json')
         with open(json_file) as file:
             data = file.read()  # Reads file contents into a string
-            render = self.summaryCareRecord.populate_template_with_json_string(data)
+            render = self.gp_summary_upload_templator.populate_template_with_json_string(data)
             XmlUtilities.assert_xml_equal(expected_string, render)
+
+    def test_hl7_parsing_returns_correct_json(self):
+        """
+        Simple assertion of correct values returned from the response parsing method
+        """
+        input_xml = FileUtilities.get_file_string(str(self.xmlFileDir / 'parseSuccessResponse.xml'))
+        parsed_data = self.gp_summary_upload_templator.parse_response(input_xml)
+
+        self.assertEqual(parsed_data['messageRef'], '9C534C19-C587-4463-9AED-B76F715D3EA3')
+        self.assertEqual(parsed_data['messageId'], '2E372546-229A-483F-9B11-EF46ABF3178C')
+        self.assertEqual(parsed_data['creationTime'], '20190923112609')
+        self.assertEqual(parsed_data['messageDetail'], 'GP Summary upload successful')
+
+    def test_hl7_parsing_bad_attribute(self):
```

Comment:
these tests should be worded based on behaviour. 

should_return_error_response_when_required_attribute_missing (etc)